src/ui/ResponsiveLayoutManager.cpp

```
@@ -0,0 +1,301 @@
+    qreal newAspectRatio = static_cast<qreal>(newWidth) / static_cast<qreal>(newHeight);
+    
+    // Calculate new values
+    QString newBreakpoint = calculateBreakpoint(effectiveHeight);
+    qreal baseLayoutScale = calculateBaseLayoutScale(effectiveHeight, newBreakpoint);
+    int newGridColumns = calculateGridColumns(newBreakpoint, newAspectRatio);
+    QString newSidebarMode = calculateSidebarMode(newBreakpoint);
+    
+    // Apply manual DPI scale override from ConfigManager
+    qreal manualOverride = 1.0;
+    ConfigManager* configManager = ServiceLocator::tryGet<ConfigManager>();
+    if (configManager) {
```

Cppcheck (2.19.0)

[error] There is an unknown macro here somewhere. Configuration is required. If slots is a macro then please configure it.

(unknownMacro)